src/core/retargeting_engine/python/retargeters/tensor_reorderer.py

```
@@ -101,9 +106,16 @@ def input_spec(self) -> RetargeterIOType:
                 )
             else:
-                inputs[input_name] = TensorGroupType(
+                tgt = TensorGroupType(
                     input_name, [FloatType(name) for name in element_names]
                 )
+
+            if tgt.is_optional:
+                raise TypeError(
+                    f"TensorReorderer does not support optional inputs, "
+                    f"but '{input_name}' is optional"
+                )
```

Ruff (0.15.2)

[warning] Avoid specifying long messages outside the exception class

(TRY003)